services/math.py

```
@@ -0,0 +1,281 @@
+        vars_dict = variables or {}
+
+        if vars_dict:
+            # Substitute variable names with their values
+            substituted = expression
+            for name, value in vars_dict.items():
+                substituted = substituted.replace(name, str(value))
+            expression = substituted
+
+        self._expr_stack = []
+        try:
+            parsed = self._bnf.parseString(expression, parseAll=True)
```

Flake8 (7.3.0)

[error] local variable 'parsed' is assigned to but never used

(F841)